### PR TITLE
Web3mail

### PR DESCRIPTION
**Pull request by 0xRomain**

### CODE REVIEW

File: src/modules/Web3mail/context/web3mail.tsx [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+      // const provider2 = await getAccount().connector?.getProvider();
+      // setDataProtector(new IExecDataProtector(provider2));
+      // setWeb3mail(provider2);
+    };
+    fetchData();
+  }, [account]);
+
+  /*
+   * @what: Check if platform has access to the user email
+   * @when: Execute it only once after user is connected, and only if user switch wallet
```
Comment (quent043) [inline src/modules/Web3mail/context/web3mail.tsx:88]:
```suggestion
   * @when: Execute it only once after the user is connected, and only if the user switches wallet
```

File: src/modules/Web3mail/context/web3mail.tsx [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    const fetchData = async () => {
+      log('Web3MailProvider ----  before fetching');
+      const protectedData = await dataProtector.fetchProtectedData({
+        owner: account.address,
+        requiredSchema: {
+          email: 'string',
+        },
+      });
+
+      const protectedEmail = protectedData.find(item => item.name === 'Untitled');
```
Comment (quent043) [inline src/modules/Web3mail/context/web3mail.tsx:109]: Why keep "Untitled ?" Shouldn"t we set a TL Standard for this ?

File: src/modules/Web3mail/context/web3mail.tsx [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+            email: email,
+          },
+        });
+
+        log('Web3MailProvider ---- protectEmailAndGrantAccess', {
+          newProtectedEmail,
+        });
+      } else {
+        log(
+          'Web3MailProvider ---- protectEmailAndGrantAccess --- email is already protected just need to grant',
```
Comment (quent043) [inline src/modules/Web3mail/context/web3mail.tsx:183]:
```suggestion
          'Web3MailProvider ---- protectEmailAndGrantAccess --- email is already protected just need to grant access',
```

File: src/pages/dashboard/profile/edit/trust-score.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    <Layout>
+      <h2 className='text-white text-xl font-bold text-center my-4'>Comming soon</h2>
+      <div className='bg-gray-200 relative flex flex-1 items-center justify-center bg-gradient-to-br p-5 shadow-xl rounded-xl'>
+        <div className='relative z-20 flex flex-col gap-3'>
+          <p className=''>
+            <span className='text-gray-800'>
+              {' '}
+              Gain trust in the network
+              <br />
+              Certified your existing web3 reputation with blockchain
```
Comment (quent043) [inline src/pages/dashboard/profile/edit/trust-score.tsx:15]:
```suggestion
              Certifiy your existing web3 reputation with blockchain
```

File: src/modules/Web3mail/context/web3mail.tsx (hunk 2 above)
Comment (0xRomain) [reply src/modules/Web3mail/context/web3mail.tsx:109]: it's just to be more easy to the user, one wallet, one default shared email

**Pull request merged**